lib_eio_linux/eio_stubs.c

```diff
@@ -34,16 +34,15 @@ CAMLprim value caml_eio_mkdirat(value v_fd, value v_path, value v_perm) {
 CAMLprim value caml_eio_getrandom(value v_ba, value v_off, value v_len) {
   CAMLparam1(v_ba);
   ssize_t ret;
-  size_t off = (size_t)Long_val(v_off);
-  size_t len = (size_t)Long_val(v_len);
-  while (off < len) {
+  ssize_t off = (ssize_t)Long_val(v_off);
+  ssize_t len = (ssize_t)Long_val(v_len);
+  do {
     void *buf = Caml_ba_data_val(v_ba) + off;
     caml_enter_blocking_section();
     ret = getrandom(buf, len - off, 0);
     caml_leave_blocking_section();
-    if (ret < 0 && errno == EINTR) continue;
-    else if (ret < 0) uerror("getrandom", Nothing);
     off += ret;
```

Comment:
This shouldn't be here. Either `off` is never used again, or `ret` is negative.

Reply:
done

Reply:
It's still there. Did git-push fail?

Reply:
Hmmm ... it not there.